### ISSUE TITLE
IndexError when accessing API reference "/api"

### ISSUE THREAD

**Issue by huberemanuel** [gradio-app/gradio#1478]
### Describe the bug

When accessing the API reference URL (a.k.a `localhost:PORT/api`) the following error is thrown:
```
Traceback (most recent call last):
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 366, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/routing.py", line 227, in app
    raw_response = await run_endpoint_function(
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/gradio/routes.py", line 234, in api_docs
    input_types_doc, input_types = get_types(inputs, "input")
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/gradio/routes.py", line 339, in get_types
    docset.append(doc_lines[1].split(":")[-1])
IndexError: list index out of range
```

With previous versions of gradio it was possible to access the API specification through this URL.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Run the getting started app: 
```
import gradio as gr

def greet(name):
    return "Hello " + name + "!!"

demo = gr.Interface(fn=greet, inputs="text", outputs="text")

demo.launch()

```
In terminal:
`python app.py`

Access the app URL with `/api` at the end.

### Screenshot

![Screenshot from 2022-06-07 11-13-23](https://user-images.githubusercontent.com/6238718/172402489-b554c8a9-e066-4aa3-9849-b08c81e82713.png)

### Logs

```shell
Running on local URL:  http://127.0.0.1:7860/

To create a public link, set `share=True` in `launch()`.
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 366, in run_asgi
    result = await app(self.scope, self.receive, self.send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 75, in __call__
    return await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/applications.py", line 269, in __call__
    await super().__call__(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/middleware/cors.py", line 84, in __call__
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/exceptions.py", line 93, in __call__
    raise exc
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/exceptions.py", line 82, in __call__
    await self.app(scope, receive, sender)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 670, in __call__
    await route.handle(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 266, in handle
    await self.app(scope, receive, send)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/routing.py", line 65, in app
    response = await func(request)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/routing.py", line 227, in app
    raw_response = await run_endpoint_function(
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/fastapi/routing.py", line 162, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/gradio/routes.py", line 234, in api_docs
    input_types_doc, input_types = get_types(inputs, "input")
  File "/home/huber/Desktop/test/env/lib/python3.8/site-packages/gradio/routes.py", line 339, in get_types
    docset.append(doc_lines[1].split(":")[-1])
IndexError: list index out of range
```

### System Info

```shell
Gradio version: 3.0.12
uname -a: Linux 5.13.0-44-generic #49~20.04.1-Ubuntu SMP Wed May 18 18:44:28 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Severity

blocking upgrade to latest gradio version

**Comment by abidlabs** [comment]
Thanks for letting us know @huberemanuel. I think the fix should be in pretty soon, right @aliabd?

**Comment by abidlabs** [comment]
Hi @huberemanuel this should be fixed (you'll need to upgrade to `gradio==3.0.13`, but what you'll find now is that there is a "view API" link in the footer which opens up the API docs in the same page (as opposed to a separate /api page)

Here's what it looks like:

<img width="644" alt="image" src="https://user-images.githubusercontent.com/1778297/172496557-c9d294b1-e2b9-46ad-9112-759fb1871f31.png">

And when you click on it:

<img width="600" alt="image" src="https://user-images.githubusercontent.com/1778297/172496572-93e1b619-8214-4648-94d6-9de04a8968c3.png">
